src/main/scala/loamstream/drm/DrmStatus.scala

```diff
@@ -50,58 +41,42 @@ object DrmStatus {
-    override def withResources(rs: DrmResources): DrmStatus = copy(resourcesOpt = Option(rs))
-  }
-  
-  final case class DoneUndetermined(override val resourcesOpt: Option[DrmResources] = None) extends DrmStatus {
-    override def withResources(rs: DrmResources): DrmStatus = copy(resourcesOpt = Option(rs))
-  }
+  final case class CommandResult(exitStatus: Int) extends DrmStatus
   
-  final case class Suspended(override val resourcesOpt: Option[DrmResources] = None) extends DrmStatus {
-    override def withResources(rs: DrmResources): DrmStatus = copy(resourcesOpt = Option(rs))
-  }
-  
-  final case class Undetermined(override val resourcesOpt: Option[DrmResources] = None) extends DrmStatus {
-    override def withResources(rs: DrmResources): DrmStatus = copy(resourcesOpt = Option(rs))
-  }
+  case object Failed extends DrmStatus
+  case object DoneUndetermined extends DrmStatus
+  case object Suspended extends DrmStatus
+  case object Undetermined extends DrmStatus
 
   import Session._
 
   def fromDrmStatusCode(status: Int): DrmStatus = status match {
     case QUEUED_ACTIVE                                              => Queued
     case SYSTEM_ON_HOLD | USER_ON_HOLD | USER_SYSTEM_ON_HOLD        => QueuedHeld
     case RUNNING                                                    => Running
-    case SYSTEM_SUSPENDED | USER_SUSPENDED | USER_SYSTEM_SUSPENDED  => Suspended()
+    case SYSTEM_SUSPENDED | USER_SUSPENDED | USER_SYSTEM_SUSPENDED  => Suspended
     case DONE                                                       => Done
-    case FAILED                                                     => Failed()
-    case UNDETERMINED | _                                           => Undetermined()
+    case FAILED                                                     => Failed
+    case UNDETERMINED | _                                           => Undetermined
   }
 
   def toJobStatus(status: DrmStatus): JobStatus = status match {
-    case Done                                                       => JobStatus.WaitingForOutputs
-    case CommandResult(exitStatus, _)                               => JobStatus.fromExitCode(exitStatus)
-    case DoneUndetermined(resources)                                => JobStatus.Failed
-    case Failed(resources)                                          => JobStatus.Failed
-    case Queued | QueuedHeld | Requeued | RequeuedHeld              => JobStatus.Submitted
-    case Running                                                    => JobStatus.Running
-    case Suspended(resources)                                       => JobStatus.Failed
-    case Undetermined(resources)                                    => JobStatus.Unknown
+    case Done                                          => JobStatus.WaitingForOutputs
+    case CommandResult(exitStatus)                     => JobStatus.fromExitCode(exitStatus)
+    case DoneUndetermined                              => JobStatus.Failed
```

Comment:
The 3 possible "done" states here make me think that the previous comment about `isFinished` may not be enough - or that I don't have a complete picture here to make an educated comment. ;)

Reply:
It's tricky.  These are all states that Uger and LSF can put a job in.  I don't know under what circumstances a job can be "done" but not have an exit code (maybe if it's declared a failure before it even starts running?) but it's definitely possible and needs to be handled. :\

That said, I can't see what `DoneUndetermined` is for, or if it's even used.  I may be able to get rid of that one.